core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/UpxCompressionBuildStep.java

```diff
@@ -0,0 +1,204 @@
+package io.quarkus.deployment.pkg.steps;
+
+import static io.quarkus.deployment.pkg.steps.LinuxIDUtil.getLinuxID;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.apache.commons.lang3.RandomStringUtils;
+import org.apache.commons.lang3.SystemUtils;
+import org.jboss.logging.Logger;
+
+import io.quarkus.deployment.annotations.BuildProducer;
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.pkg.NativeConfig;
+import io.quarkus.deployment.pkg.builditem.ArtifactResultBuildItem;
+import io.quarkus.deployment.pkg.builditem.NativeImageBuildItem;
+import io.quarkus.deployment.util.FileUtil;
+import io.quarkus.deployment.util.ProcessUtil;
+
+public class UpxCompressionBuildStep {
+
+    private static final Logger log = Logger.getLogger(UpxCompressionBuildStep.class);
+
+    /**
+     * The name of the environment variable containing the system path.
+     */
+    private static final String PATH = "PATH";
+
+    @BuildStep(onlyIf = NativeBuild.class)
+    public void compress(NativeConfig nativeConfig, NativeImageBuildItem image,
+            BuildProducer<ArtifactResultBuildItem> artifactResultProducer) {
+        if (nativeConfig.compression.level.isEmpty()) {
+            log.debug("UPX compression disabled");
+            return;
+        }
+
+        Optional<File> upxPathFromSystem = getUpxFromSystem();
+        boolean inContainerUpx = upxPathFromSystem.isEmpty() && nativeConfig.isContainerBuild();
```

Comment:
```suggestion
        boolean inContainerUpx = upxPathFromSystem.isEmpty() || nativeConfig.isContainerBuild();
```

It works if I do this 👆🏻